router/pkg/config/config.go

```diff
@@ -398,20 +398,28 @@ type OverridesConfiguration struct {
 	Subgraphs map[string]SubgraphOverridesConfiguration `yaml:"subgraphs"`
 }
 
-type AuthenticationProviderJWKS struct {
-	URL                 string        `yaml:"url"`
-	HeaderNames         []string      `yaml:"header_names"`
-	HeaderValuePrefixes []string      `yaml:"header_value_prefixes"`
-	RefreshInterval     time.Duration `yaml:"refresh_interval" envDefault:"1m"`
+type JWKSConfiguration struct {
+	URL             string        `yaml:"url"`
+	Algorithms      []string      `yaml:"algorithms"`
+	RefreshInterval time.Duration `yaml:"refresh_interval" envDefault:"1m"`
 }
 
-type AuthenticationProvider struct {
-	Name string                      `yaml:"name"`
-	JWKS *AuthenticationProviderJWKS `yaml:"jwks"`
+// TODO Change name to TokenSource maybe?
+type HeaderSource struct {
+	Type        string `yaml:"type"`
+	Name        string `yaml:"name"`
+	ValuePrefix string `yaml:"value_prefix"`
+}
```

Comment:
The code allows to specify multiple `HeaderSources` with the same name and a different value prefix. Maybe we could also make this a list of value prefixes if someone decides that they want to have multiple options for prefixes